src/hooks/useAuthorizationRoles.js

```diff
@@ -1,23 +1,35 @@
 import { useQuery } from 'react-query';
 
-import { useNamespace, useOkapiKy } from '@folio/stripes/core';
+import { useOkapiKy, useNamespace, useStripes } from '@folio/stripes/core';
 
+import { useEffect, useState } from 'react';
 import { ROLES_ENDPOINT } from '../constants/endpoints';
 
-const useAuthorizationRoles = ({ searchTerm, options }) => {
+const useAuthorizationRoles = () => {
   const ky = useOkapiKy();
+  const [namespace] = useNamespace({ key: 'ui-authorization-roles' });
+  const stripes = useStripes();
+  const [roles, setRoles] = useState([]);
 
-  const [nameSpace] = useNamespace({ key: 'ui-authorization-roles' });
-
-  const { data, isLoading, refetch } = useQuery(
-    [nameSpace],
-    () => ky.get(ROLES_ENDPOINT(searchTerm)).json(),
-    { enabled: true, ...options }
+  const { data, isLoading, refetch, isSuccess } = useQuery(
+    namespace,
+    () => ky.get(ROLES_ENDPOINT(stripes.config.maxUnpagedResourceCount)).json()
   );
 
+  const filterRoles = (searchTerm) => {
+    setRoles(data.roles.filter(role => role.name.toLowerCase().includes(searchTerm.toLowerCase())));
+  };
+
+  useEffect(() => {
+    if (isSuccess) {
+      setRoles(data?.roles || []);
+    }
+  }, [data, isSuccess]);
+
   return {
-    roles: data?.roles || [],
+    roles,
     isLoading,
+    filterRoles,
```

Comment:
Can you provide a suggestion, please? 
Currently, data.roles are immutable, local state `roles` is created. On isSuccess it sets roles to local state. filterRoles is to make filtration on the client, just taking initial data.roles and 
 return filtered roles by `searchTerm`. I'm not sure if it is necessary to cloneDeep data.roles

Reply:
I think Zak's suggestion is to just do 
```
 const filterRoles = (searchTerm) => {
    return roles.filter(role => role.name.toLowerCase().includes(searchTerm.toLowerCase()));
  };
```

I agree with Zak on treating `roles` as immutable rather than relying on `data.roles`.  You can always use a variable in the component to manage the return value that is currently `roles`, but I think there should only be a call to `setRoles` after the API call has completed.

Reply:
And you mean to create local state of roles in the component itself?